### ISSUE TITLE
Make --maskSecrets the default behavior and add --no-maskSecrets option

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1067]
**Problem**
Currently, secretlint outputs actual secret values by default, which poses security risks in several scenarios:

1. **Terminal Output**: Secret values remain in terminal output and can be accidentally shared
2. **CI/CD Logs**: Build logs may contain actual secrets, making them visible to anyone with access to the CI system
3. **AI Agent Tools Security Risk**: Modern AI agent tools often read terminal output and may inadvertently send sensitive data to external services

**Proposed Solution**
- Make `--maskSecrets` the default behavior
- Add a new `--no-maskSecrets` option to disable masking when needed
- This ensures secrets are masked by default while maintaining flexibility for users who need to see actual values

**Motivation**
- **Security by default**: Users shouldn't need to remember to add `--maskSecrets` to protect sensitive data
- **CI/CD Safety**: Build logs will be safe by default without additional configuration
- **Agent Tool Protection**: Prevents accidental data leakage to AI agents or other tools that monitor terminal output
- **Backward Compatibility**: Users who need actual values can explicitly use `--no-maskSecrets`

**Benefits**
- Improved security posture out of the box
- Reduced risk of accidental secret exposure
- Protection against emerging threats from AI agent tools
- Maintains existing functionality with explicit opt-out
- **No impact on user workflow**: Line numbers, file paths, and rule names remain visible for debugging and fixing issues

**User Experience**
The change maintains full usability since all essential information for developers remains available:
- File paths and line numbers for locating issues
- Rule names and descriptions for understanding violations
- Error context and surrounding code (if applicable)
- Only the actual secret values are masked, preserving all debugging information

**Implementation Notes**
- This is a breaking change that should be documented in migration guide
- Consider adding a deprecation warning in current version before making the change
- Update documentation and examples to reflect new default behavior

**Examples**

Current behavior:
```bash
secretlint file.js
# Outputs: Found secret: sk-1234567890abcdef
```

Proposed default behavior:
```bash
secretlint file.js
# Outputs: Found secret: ***

secretlint --no-maskSecrets file.js
# Outputs: Found secret: sk-1234567890abcdef
```
